### ISSUE TITLE
Not compatible with winston-log-and-exit.

### ISSUE THREAD

**Issue by jdthorpe** [imsky/winston-rsyslog2#9]
Hello, 

I'm the author of the [winston-log-and-exit](https://www.npmjs.com/package/winston-log-and-exit), and have some users of this package have complained that this package doesn't work with `winston-log-and-exit`.    As you may know, cleanly exiting a process and retaining the last few log entries is [a perennial problem for winston users](https://github.com/winstonjs/winston/issues/228), which is confounded by poor documentation on how Winston identifies streaming loggers, and what API streaming loggers should implement.    

In short, to work with `winston-log-and-exit` a custom Streaming Transport needs a [`._stream`  attribute](https://github.com/winstonjs/winston/issues/228), has an [`.end()` method](https://github.com/jdthorpe/winston-log-and-exit/blob/master/index.js#L24), and which issues a ["finish" event](https://github.com/jdthorpe/winston-log-and-exit/blob/master/index.js#L15) when all the messages that were logged before the `.end()` method was called.

Cheers,

-Jason

**Comment by imsky** [comment]
hi @jdthorpe - i'm no longer using this package (no longer using rsyslog for logs), so if you'd like to add a patch, let me know and i'll add you as a collaborator.
